test/health-monitor.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+
+function containerd_monitoring {
+  while [ 1 ]; do
+    if ! timeout 60 ctr containers ls > /dev/null; then
+      echo "Containerd failed!"
+      pkill containerd
+      # Wait for a while, as we don't want to kill it again before it is really up.
+      sleep 60
+    else
+      sleep "${SLEEP_SECOMDS}"
+    fi
+  done
+}
+
+function cri_containerd_mmonitoring {
+  while [ 1 ]; do
+    if ! timeout 60 crictl --runtime-endpoint /var/run/cri-containerd.sock ps > /dev/null; then
```

Comment:
crictl should already have it's own timeout I think

Reply:
Seems have no timeout in `crictl`, see:
https://github.com/kubernetes-incubator/cri-tools/blob/master/cmd/crictl/container.go#L548-L655

Reply:
`crictl` only have timeout of connecting to server (default: 10s), but we still need to the timeout that  if the connecting to server is ok, but the request of `crictl ps` is blocked.